sdk/logs/src/test/java/io/opentelemetry/sdk/logs/AnyValueBodyTest.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.logs;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat;
+
+import io.opentelemetry.api.logs.Logger;
+import io.opentelemetry.extension.incubator.logs.AnyValue;
+import io.opentelemetry.extension.incubator.logs.ExtendedLogRecordBuilder;
+import io.opentelemetry.extension.incubator.logs.KeyAnyValue;
+import io.opentelemetry.sdk.logs.data.Body;
+import io.opentelemetry.sdk.logs.export.SimpleLogRecordProcessor;
+import io.opentelemetry.sdk.logs.internal.AnyValueBody;
+import io.opentelemetry.sdk.testing.exporter.InMemoryLogRecordExporter;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedHashMap;
+import org.junit.jupiter.api.Test;
+
+class AnyValueBodyTest {
+
+  @Test
+  @SuppressWarnings("DoubleBraceInitialization")
+  void anyValueBody() {
+    InMemoryLogRecordExporter exporter = InMemoryLogRecordExporter.create();
+    SdkLoggerProvider provider =
+        SdkLoggerProvider.builder()
+            .addLogRecordProcessor(SimpleLogRecordProcessor.create(exporter))
+            .build();
+    Logger logger = provider.get(AnyValueBodyTest.class.getName());
+
+    // AnyValue can be a primitive type, like a string, long, double, boolean
+    extendedLogRecordBuilder(logger).setBody(AnyValue.of(1)).emit();
+    assertThat(exporter.getFinishedLogRecordItems())
+        .hasSize(1)
+        .satisfiesExactly(
+            logRecordData -> {
+              assertThat(logRecordData.getBody().getType()).isEqualTo(Body.Type.ANY_VALUE);
+              assertThat(logRecordData.getBody().asString()).isEqualTo("1");
```

Comment:
I think that the fact that you have to go through `asString()` to do this assertion might be highlighting a small shortcoming. Ideally you could just get the body as an `AnyValue`, but because `LogRecordData.getBody()` always returns `Body` and not ever `AnyValueBody` (or at least the caller can't tell based on the interface), this is still necessary. Do I have that right?

I haven't thought this through, but what if we added a default `Body.getAnyBody()` that returned `AnyValue.of(asString())`? Does that feel misplaced? 

Might another option could be to have `LogRecordData.getAnyBody()`? I dunno, I just think we should try and improve this.

Reply:
I think when we're ready to commit to this API, we update `Body` with a `getAnyValue` method. This cast shenanigans is how I think we can publish a working version of this workflow while still retaining the ability to make breaking changes in response to feedback.

Once `Body#getAnyValue()` exists, its a lot cleaner. 

> I haven't thought this through, but what if we added a default Body.getAnyBody() that returned AnyValue.of(asString())? Does that feel misplaced?

No that's what I envision as well.

Reply:
Cool, seems reasonable to me. 😎